app/src/main/scala/org/alephium/explorer/persistence/queries/OutputQueries.scala

```diff
@@ -113,6 +113,22 @@ object OutputQueries {
     }
   }
 
+  // format: off
+  def outputsFromTxsNoJoin(txHashes: Seq[Transaction.Hash]):
+  DBActionR[Seq[(Transaction.Hash, Int, Int, Hash, U256, Address, Option[TimeStamp], Option[Transaction.Hash])]] = {
+  // format: on
+    if (txHashes.nonEmpty) {
+      val values = txHashes.map(hash => s"'\\x$hash'").mkString(",")
+      sql"""
+    SELECT outputs.tx_hash, outputs.output_order, outputs.hint, outputs.key,  outputs.amount, outputs.address, outputs.lock_time, outputs.spent_finalized
+    FROM outputs
+    WHERE outputs.tx_hash IN (#$values) AND outputs.main_chain = true
+    """.as
+    } else {
+      DBIOAction.successful(Seq.empty)
+    }
+  }
```

Comment:
You reckon there is any benefit in using prepared statement here and for `updateTransactionPerAddresses` in `InputUpdateQueries`? Something like the following? 

```scala
val placeholder = paramPlaceholder(rows = 1, columns = txHashes.size)

val query =
  s"""
  SELECT outputs.tx_hash, outputs.output_order, outputs.hint, outputs.key,  outputs.amount, outputs.address, outputs.lock_time, outputs.spent_finalized
  FROM outputs
  WHERE outputs.tx_hash IN $placeholder AND outputs.main_chain = true"""

val parameters: SetParameter[Unit] =
  (_: Unit, params: PositionedParameters) =>
    txHashes foreach { output =>
      params >> output
  }

SQLActionBuilder(
  queryParts = query,
  unitPConv  = parameters
).as
```
I can confirm that different queries with different parameter sizes use different prepared statements so chances of the same prepared statement being used again for randomly sized inputs is lowered. Although this seems odd because it probably shouldn't. I will try to see if this can be avoided somehow.  

``` 
//Prepared-statement with 11 hashes
(S_3,2022-06-30 13:47:48.789585+10,{bytea,bytea,bytea,bytea,bytea,bytea,bytea,bytea,bytea,bytea,bytea},
  SELECT outputs.tx_hash, outputs.output_order, outputs.hint, outputs.key,  outputs.amount, outputs.address, outputs.lock_time, outputs.spent_finalized
  FROM outputs
  WHERE outputs.tx_hash IN ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) AND outputs.main_chain = true)

//Prepared-statement with 2 hashes
(S_7,2022-06-30 13:47:49.161239+10,{bytea,bytea},
  SELECT outputs.tx_hash, outputs.output_order, outputs.hint, outputs.key,  outputs.amount, outputs.address, outputs.lock_time, outputs.spent_finalized
  FROM outputs
  WHERE outputs.tx_hash IN ($1, $2) AND outputs.main_chain = true)
```

Reply:
ah yes I always forgot we can use this, it's much cleaner. But while I'm writing this, I remember there is a size limit of placeholders no (not sure at all)? what if we have a lot of tx_hashes?

Reply:
Oh good point. We could add a `QuerySplitter.splitIn(...)` but I guess that would be alarming as well because if we have to run such large `select` queries with multiple 1000's (> `Short.MaxValue`) of parameters within the `IN` clause then we maybe need to adjust the schema a little? Probably needs grouping into a single logical unique id instead of having to use the `IN` clause.

Sorry for a very vague reply. With the whole 'double checking' tasks I'm sorta looking at this kinda stuff (schema related). Will have a better idea as that progresses.

Reply:
actually the number of tx_hash in the `IN` will be limited to [100](https://github.com/alephium/explorer-backend/blob/bee50e537607fd808c8b7eee8c49fd819bbe229d/app/src/main/scala/org/alephium/explorer/api/model/Pagination.scala#L25). But maybe it's safer to come with a safer version? as you said it would be alarming if we have > 1000 entries for a select.

Reply:
You are right, would be good to be safer but I guess then this would also apply to all our existing queries (around 8 queries) that use `IN` or `NOT IN` clause. 

Maybe something to work on in an independent issue?

Reply:
yep, I created https://github.com/alephium/explorer-backend/issues/288